### PR TITLE
Fix write when using transaction context

### LINKED ISSUE #531: AttributeError: '_io.BufferedReader' object has no attribute 'temp'
Hello,

I try to use `LocalFileSystem` with Transaction context manager and I get the following error:

```
AttributeError: '_io.BufferedReader' object has no attribute 'temp'
```

An example:

```python
from fsspec.implementations.local import LocalFileSystem

def main():
    fs = LocalFileSystem()

    with fs.transaction:
        with fs.open('test.txt', 'w') as fp:
            fp.write('hello')

        with fs.open('test.txt', 'r') as fp:
            s = fp.read()
        
        print(s)

if __name__ == '__main__':
    main()
```

An exception:

```
Traceback (most recent call last):
  File "scratches\scratch_94.py", line 18, in <module>
    main()
  File "scratches\scratch_94.py", line 14, in main
    print(s)
  File "lib\site-packages\fsspec\transaction.py", line 24, in __exit__
    self.complete(commit=exc_type is None)
  File "lib\site-packages\fsspec\transaction.py", line 37, in complete
    f.commit()
  File "lib\site-packages\fsspec\implementations\local.py", line 246, in commit
    os.replace(self.temp, self.path)
  File "lib\site-packages\fsspec\implementations\local.py", line 261, in __getattr__
    return getattr(self.f, item)
AttributeError: '_io.BufferedReader' object has no attribute 'temp'
```

If I only use write operations within transaction context it works but read operation leads to the exception.

### PR DESCRIPTION
**Pull request by andersy005**
Fixes #531

### CODE REVIEW

**Comment by martindurant** [comment]
This seems to be failing lint, but the gha details box is bizarrely empty

**Comment by andersy005** [comment]
> but the gha details box is bizarrely empty 

GitHub is experiencing [some issues](https://www.githubstatus.com/) 

> This seems to be failing lint,

I used the pre-commit hooks locally. I am noticing that CI runs black and flake8 directly. Will fix the linting issues shortly

**Comment by martindurant** [comment]
It would be reasonable to make an effort to have the CI and commit hooks be identical. I have found that new versions of `black` often have small, subtle, annoying changes.

**Comment by andersy005** [comment]
@martindurant, unless I am missing something, the failing tests seem unrelated to changes in this PR

File: fsspec/implementations/tests/test_local.py [hunk 1]
```diff
@@ -597,7 +597,7 @@ def test_copy_errors(tmpdir):
 
 
 def test_transaction(tmpdir):
-    file = (tmpdir / "test.txt").as_posix()
+    file = str(tmpdir / "test.txt")
```
Comment (martindurant) [inline fsspec/implementations/tests/test_local.py:600]: Would this do the right thing on Windows?

Comment (espdev) [reply fsspec/implementations/tests/test_local.py:600]: I can confirm that on windows we need to use as_posix path representation. I just recently fixed such issue in my code.

**Comment by martindurant** [comment]
Test passes also on windows, so I will merge. This is actually a good demonstration that fsspec is doing the right thing with windows, at least for local paths.

**Pull request merged**